packages/@react-aria/utils/src/useSyntheticBlurEvent.ts

```diff
@@ -0,0 +1,205 @@
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {FocusEvent as ReactFocusEvent, useCallback, useRef} from 'react';
+import {useLayoutEffect} from './useLayoutEffect';
+
+class SyntheticBlurEvent implements ReactFocusEvent {
+  nativeEvent: FocusEvent;
+  target: Element;
+  currentTarget: Element;
+  relatedTarget: Element;
+  bubbles: boolean;
+  cancelable: boolean;
+  defaultPrevented: boolean;
+  eventPhase: number;
+  isTrusted: boolean;
+  timeStamp: number;
+  type: string;
+
+  constructor(type: string, nativeEvent: FocusEvent) {
+    this.nativeEvent = nativeEvent;
+    this.target = nativeEvent.target as Element;
+    this.currentTarget = nativeEvent.currentTarget as Element;
+    this.relatedTarget = nativeEvent.relatedTarget as Element;
+    this.bubbles = nativeEvent.bubbles;
+    this.cancelable = nativeEvent.cancelable;
+    this.defaultPrevented = nativeEvent.defaultPrevented;
+    this.eventPhase = nativeEvent.eventPhase;
+    this.isTrusted = nativeEvent.isTrusted;
+    this.timeStamp = nativeEvent.timeStamp;
+    this.type = type;
+  }
+
+  isDefaultPrevented(): boolean {
+    return this.nativeEvent.defaultPrevented;
+  }
+
+  preventDefault(): void {
+    this.defaultPrevented = true;
+    this.nativeEvent.preventDefault();
+  }
+
+  stopPropagation(): void {
+    this.nativeEvent.stopPropagation();
+    this.isPropagationStopped = () => true;
+  }
+
+  isPropagationStopped(): boolean {
+    return false;
+  }
+
+  persist() {}
+}
+
+export function useSyntheticBlurEvent(onBlur: (e: ReactFocusEvent) => void) {
+  let stateRef = useRef({
+    isFocused: false,
+    onBlur,
+    observer: null as MutationObserver,
+    target: null as HTMLElement
+  });
+  stateRef.current.onBlur = onBlur;
+
+  // Clean up MutationObserver on unmount. See below.
+  // eslint-disable-next-line arrow-body-style
+  useLayoutEffect(() => {
+    const state = stateRef.current;
+    return () => {
+      if (stateRef.current.target) {
+        stateRef.current.target.removeEventListener('focusout', onBlurHandler);
+      }
+      if (state.observer) {
+        state.observer.disconnect();
+        state.observer = null;
+      }
+    };
+  }, []);
+
+
+  let onBlurHandler = useCallback((e: FocusEvent) => {
+    let target = e.target;
+    stateRef.current.isFocused = false;
+
+    if (((target instanceof HTMLButtonElement ||
+      target instanceof HTMLInputElement ||
+      target instanceof HTMLTextAreaElement ||
+      target instanceof HTMLSelectElement) && target.disabled) || !document.body.contains(e.target as HTMLElement)) {
```

Comment:
Whats the significance behind these specific elements? I'm guessing this is all the focusable elements that can be disabled?
(also you can replace your `e.target` with `target`)

Reply:
These are elements that, when disabled, do not fire blur. logic hasn't changed from previous iteration in that regard https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/interactions/src/utils.ts#L111
just also added the case where the element was unmounted